core/sql/exp/exp_eval.cpp

```diff
@@ -862,7 +862,17 @@ ex_expr::exp_return_type ex_expr::evalClauses(ex_clause *clause,
 	atp1->setDiagsArea(diagsArea);
 
       if (retcode == ex_expr::EXPR_ERROR)
-	return retcode;
+      {
+        if((getPCodeMode() & ex_expr::ERROR_CONTINUE) > 0)
+        {
+          ExpTupleDesc::setNullValue( op_data[0],
+                                      clause->getOperand(0)->getNullBitIndex(),
+                                      clause->getOperand(0)->getTupleFormat() );
+          retcode = ex_expr::EXPR_OK;
```

Comment:
Couple things:
Codegen time: The flag to move null in case of an error should be set only if target is nullable.
Runtime: Expr code should check to make sure that target is nullable before moving
the null value. If target is non-nullable, then original error should be returned.